cloud/pkg/devicecontroller/controller/downstream.go

```diff
@@ -308,10 +308,21 @@ func (dc *DownstreamController) sendDeviceModelMsg(device *v1beta1.Device, opera
 	if device == nil || device.Spec.DeviceModelRef == nil {
 		return
 	}
-	edgeDeviceModel, ok := dc.deviceModelManager.DeviceModel.Load(device.Spec.DeviceModelRef.Name)
-	if !ok {
-		klog.Warningf("not found device model for device: %s, operation: %s", device.Name, operation)
-		return
+	var edgeDeviceModel any
+	var ok bool
+	i := 0
+	for {
```

Comment:
Can we include a third-party retry util https://github.com/avast/retry-go, or wrap a retry util in the `pkg/util/retry` package?  @fisherxu

Reply:
Is it like this below?
err := retry.Do(
		func() error {
			edgeDeviceModel, ok = dc.deviceModelManager.DeviceModel.Load(device.Spec.DeviceModelRef.Name)
			if !ok {
				return fmt.Errorf("not found device model for device: %s, operation: %s", device.Name, operation)
			} else {
				return nil
			}
		},
		retry.Delay(1*time.Second),
		retry.Attempts(3),
		retry.DelayType(retry.FixedDelay),
	)
	if err != nil {
		klog.Warningf(err.Error())
		return
	}

Reply:
Yes, but we need to discuss whether to use a third-party package or to write our own utils.

Reply:
> Yes, but we need to discuss whether to use a third-party package or to write our own utils.

If we write utils ourselves, it may not be easy to abstract the code that needs to be retried into a function.